### DISCUSSION TITLE
One Project don't show any coverage

### DISCUSSION THREAD

**Discussion by MichaelInsberg** [FortuneN/FineCodeCoverage#287] [Q&A] [answered]
Hi, I user FCC with VS2022 and it works pretty well.

But for one project I get no Coverage in the log file I see this error message:

Unable to instrument module: C:\Projekte\Framework4.0\Build\Tests\Debug\fine-code-coverage\build-output\Core.UI.UILibrary.dll because : Failed to resolve System.Windows.Threading.DispatcherPriority

The Project file looks like this:

<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

    <PropertyGroup>
        <TargetFramework>netcoreapp3.1</TargetFramework>
        <RootNamespace>$(SolutionName).$(MSBuildProjectName)</RootNamespace>
        <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
	    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
        <UseWpf>true</UseWpf>
    </PropertyGroup>

If I use another coverage tool l see the coverage result of this project:

![image](https://user-images.githubusercontent.com/78546805/186080683-1e7f7fad-2400-4107-8525-f4dd5e486e8a.png)

But in FCC I see no coverage even in the log file the Core.UI.UILibrary is missing.

**Comment by tonyhallett** [comment, marked as answer]
I assume you are receiving coverage from coverlet.  This may help https://github.com/coverlet-coverage/coverlet/issues/1221

Alternatively you could switch to the ms code coverage option. Although there is an issue with async methods until the next release

**Reply by MichaelInsberg** [reply]
Hi I don't use coverlet coverage.

The picture is from Reshaper. But this is nocht the proble I only want to show that there are Unit tests for this project.

When I use the output from FCC I don't see Core.UI.UILibrary in the FCC report.

**Reply by tonyhallett** [reply]
You do with FCC with SDK style projects and you have not set RunMsCodeCoverage

**Reply by MichaelInsberg** [reply]
Can you send me an example for this?

**Reply by tonyhallett** [reply]
Of RunMsCodeCoverage ?

**Reply by MichaelInsberg** [reply]
I use Nunit for Unit Tests not MsUnit.

![image](https://user-images.githubusercontent.com/78546805/186162316-ecb46498-7ca7-4df5-abc4-03c4a2e2ddd5.png)

And don't have the option discribe here:

**Reply by MichaelInsberg** [reply]
[Link](https://docs.microsoft.com/de-de/visualstudio/test/using-code-coverage-to-determine-how-much-code-is-being-tested?view=vs-2022&tabs=csharp
)

**Comment by tonyhallett** [comment]
RunMsCodeCoverage is an FCC setting.
[Link](https://github.com/FortuneN/FineCodeCoverage#how-to-utilize-ms-code-coverage-with-fcc-)

**Reply by MichaelInsberg** [reply]
But I don't use MS Test and it works for all other project without this setting

**Comment by tonyhallett** [comment]
FCC does not do coverage itself.  It uses one of Coverlet, OpenCover or Microsoft.  Each of these coverage services can work with NUnit, XUnit or MSTest.

For your project Coverlet is being used, hence the first link I provided.
You can switch to Microsoft by setting the FCC option RunMsCodeCoverage.  The second link I provided.

**Reply by MichaelInsberg** [reply]
This are my settings:
![image](https://user-images.githubusercontent.com/78546805/186435176-70ae0a94-359c-4b13-8983-a01000852add.png)
![image](https://user-images.githubusercontent.com/78546805/186435277-714cbe46-ac57-4748-9fd9-f26d76d01d6b.png)
![image](https://user-images.githubusercontent.com/78546805/186435456-8fdb082c-48cc-4064-aef7-5bbce01fa6db.png)
Anything wrong?

**Reply by tonyhallett** [reply]
Probably. You have not expanded all settings.

Have you looked at the coverlet link yet ? 
Please take the advice there or switch to ms code coverage.

**Reply by MichaelInsberg** [reply]
![image](https://user-images.githubusercontent.com/78546805/186580278-4f9c306d-fbc8-442c-be33-a3a3d36afcf9.png)
This is my actual setting

**Reply by tonyhallett** [reply]
I could see that from your previous screenshot.

**Reply by MichaelInsberg** [reply]
This must be Yes?

**Reply by tonyhallett** [reply]
If you are not supplying your own runsettings then yes.  If you use Ms code coverage then until the next release async methods are not covered properly.  So I do suggest that you look at the coverlet issue link, as it is specifically about missing wpf dlls and how to rectify.

**Reply by MichaelInsberg** [reply]
Thanks set it to true seems to solve the problem
